analytics.go

```diff
@@ -453,12 +454,35 @@ func (c *client) send(msgs []message, retryAttempt int) {
 // Upload serialized batch message.
 func (c *client) upload(b []byte, targetNode string) error {
 	url := c.Endpoint + "/v1/batch"
-	req, err := http.NewRequest("POST", url, bytes.NewReader(b))
+	gzipPayload := func(data []byte) (io.Reader, error) {
+		var b bytes.Buffer
+		gz := gzip.NewWriter(&b)
+		_, err := gz.Write(data)
+		if err != nil {
```

Comment:
```suggestion
		if _, err := gz.Write(data); err != nil {
```

Reply:
This optimization is not correct.

Reply:
please explain

Reply:
If I do this:
                 if gz, err := gzip.NewWriterLevel(&b, gzip.BestSpeed); err != nil {
			return nil, err
		}
		if err = gz.Close(); err != nil {
			return nil, err
		}

gz falls is becoming undeclared variable.

Reply:
this comment was about `gz.Write`, not `gzip.NewWriterLevel`

Reply:
Oh, then it is outdated.

Reply:
Rectified.